pkg/engine/common.go

```diff
@@ -64,6 +64,11 @@ func zeroToCurrent(ctx, conn context.Context, m Method, target *Target, tag *[]s
 	return nil
 }
 
+type empty struct {
+}
+
+var badCommitList map[plumbing.Hash]empty = make(map[plumbing.Hash]empty)
```

Comment:
super nit: it _may be_  more idiomatic to see:
`var badCommitList map[plumbing.Hash]struct{} = make(map[plumbing.Hash]struct{})`